frontends/web/src/api/backend.ts

```diff
@@ -70,3 +72,7 @@ export const getDefaultConfig = (): Promise<any> => {
 export const socksProxyCheck = (proxyAddress: string): Promise<ISuccess> => {
   return apiPost('socksproxy/check', proxyAddress);
 };
+
+export const ethDecodeMessage = (message: string): Promise<TDecodedMessage> => {
```

Comment:
sorry for the late suggestion, but why not simply make this a pure JS function? Untested from ChatGPT:

```
function ethDecodeHex(message: string): string {
  let str = '';
  for (let i = 0; i < message.length; i += 2) {
    str += String.fromCharCode(parseInt(message.substr(i, 2), 16));
  }
  return decodeURIComponent(escape(str));
}
```

would that work? if so, it seems a lot simpler compared to this current PR

Reply:
I'll test out how it works with JS and I can change it. 
The reason for doing it in the backend was that you mentioned it when we first talked about it and the idea was that it would be easier/more robust just to do a simple api call and do the decoding in go

Reply:
I meant that versus pulling in more NPM deps, as we want to keep the NPM deps to a minimum. If the functionality is so simple like above I see no problem in adding the code to the frontend instead.

Reply:
escape() is deprecated: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/escape

Did it slightly differently (with GPTs help :D) and added some validation and error checking, all in the frontend now